Add new set-version task to native Makefile.toml

JJRcop opened the pull request:
New duckscript task to set the version number when building from source yourself
You can now run for example `makers set-version 2.7.2` to set the version in the native code to 2.7.2
You can also run `makers set-version` without argument to automatically fetch the closest git tag using [`git describe`](https://git-scm.com/docs/git-describe/2.41.0).

This does not cause the `build` task to depend on `set-version`. ~~Also, as it is now the task cannot update the version multiple times, and the source code must be reset to update it again. This is because I have stuck to the behavior of other scripts within the source code so far which also only can update once, but it could be improved if needed.~~

Documentation for building from source within the extension and in native/README.md are also updated.

Addresses the native portion of #193

filips123 commented:
Thanks! I tried doing this last year, but got stuck with makers' bash transpiling which didn't completely work and what to do about updating the version multiple times, and then haven't got into duckscript... It looks fine, I'll probably merge it next week when I also test this a bit.

Supporting updating the version multiple times without resetting the source code would be nice, but I don't know how to do this easily. One thing I tried previously was just to match `version = "*"` (without any specific version) inside `Cargo.toml`, but this would also match and update [version requirement for `windows`](https://github.com/filips123/PWAsForFirefox/blob/main/native/Cargo.toml#L57) and break things. Maybe it could also match for other things ([like this comment](https://github.com/filips123/PWAsForFirefox/blob/main/native/Cargo.toml#L15)) to prevent other `version`s from being changed, although this could break in case I edit/remove that comment. Another possibility would be to reset the repository (or maybe just `Cargo.lock`) with Git, but that could also cause unwanted side effects in case the user changed some things that the script would then delete.

I guess that the proper solution would be to parse TOML and only update root's `version`, but that's probably not possible with duckscipt, and complicating things with another scripting language is probably not worth it, especially since it adds another level of complexity and another build dependency. Apparently, `cargo-make` also supports some [Rust scripts](https://github.com/sagiegurari/cargo-make#rust-code)m which might be useful to do this. I might try to implement this in the future, but for now, I guess this is fine.

Another more proper solution is to wait until Cargo natively supports editing `Cargo.toml` with commands. There are already some third-party commands to do this, but I would prefer not to depend on another program that users have to install.

JJRcop commented:
The task is now in a separate file instead of inline within `Makefile.toml`. I've also updated it so it can update multiple times.

I have addressed your concern about the [version requirement for `windows`](https://github.com/filips123/PWAsForFirefox/blob/main/native/Cargo.toml#L57) by using the comment before the version declaration as part of the match.

filips123 reviewed (approved):
This also works fine now.
